client/src/components/admin/GeneralNotification/GeneralNotificationNavigationButton.tsx

```diff
@@ -0,0 +1,75 @@
+import { Box, IconButton, Tooltip } from '@mui/material';
+import { MegaphoneIcon } from '@src/components/icons/MegaphoneIcon';
+import { useToasts } from '@src/stores/ToastContext';
+import { useTranslations } from '@src/stores/TranslationContext';
+import { request } from '@src/utils/request';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+export function GeneralNotificationNavigationButton() {
+  const { tr } = useTranslations();
+  const { showToast } = useToasts();
+  const history = useHistory();
+  const [newNotifications, setNewNotifications] = useState(false);
+
+  async function refreshRecentCount() {
+    const data = (await request('/api/general-notifications/recent-count')) as {
```

Comment:
[nit] The `request` utility function should support type parameters, so it's a bit more type-safe to do `request<{ count: number }>` instead of the type assertion.